### PR TITLE
Port to 1.20.1

### PR DESCRIPTION
**Pull request by reetamb**
Mining Master has been ported to 1.20.1, with the following necessary changes:
- Gradle files and mods.toml updated
- New method of registering creative mode tabs
- GUI methods take GuiGraphics
- LivingEntity and ItemStack method changes

In addition, minor changes have been made to the Gem Forge subtitles.

### CODE REVIEW

**Comment by reetamb** [comment]
- The Lava Bucket now renders on the confirm button of the Gem Forge
- Its texture is now part of the GUI texture of the Gem Forge, allowing resource pack creators to modify it

File: build.properties [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+MINECRAFT_VERSION=1.20.1
+FORGE_VERSION=47.2.0
 MAPPINGS_CHANNEL=official
-MAPPINGS_VERSION=1.19.4
-JEI_VERSION=13.1.0.15
+MAPPINGS_VERSION=1.20.1
+JEI_VERSION=15.3.0.4
 # Removed for now as no 1.19 versions exist
 TETRA_FILE=
 MUTIL_FILE=
```
Comment (SwanX1) [inline build.properties:9]: we should probably reimplement tetra support... maybe...
this is not related to the PR, please ignore

File: src/main/java/org/infernalstudios/miningmaster/init/MMSounds.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public class MMSounds {
 
     // GEM FORGE
     public static final RegistryObject<SoundEvent> GEM_FORGE_COOK = registerSound("gemforge.cook");
-
+    public static final RegistryObject<SoundEvent> GEM_FORGE_COMPLETE = registerSound("gemforge.complete");
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/miningmaster/init/MMSounds.java:32]: is this a bugfix or what?

File: src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java [hunk 3]
```diff
@@ -61,14 +62,20 @@ public GemSmithingRecipe(ResourceLocation recipeId, Ingredient blacklist, ItemSt
...
+        return !this.blacklist.test(inv.getItem(1)) && ItemStack.isSameItem(this.gem, inv.getItem(2)) && assemble(inv, worldIn.registryAccess()) != null;
     }
 
     @Override
     public ItemStack assemble(Container inv, RegistryAccess registryAccess) {
-        ItemStack itemstack = inv.getItem(0).copy();
-        CompoundTag compoundnbt = inv.getItem(0).getTag();
+        MiningMaster.LOGGER.debug("Slot 0: " + inv.getItem(0).copy());
+        MiningMaster.LOGGER.debug("Slot 1: " + inv.getItem(0).copy());
+        MiningMaster.LOGGER.debug("Slot 2: " + inv.getItem(0).copy());
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java]: these log statements should probably be `trace()` not `debug()`, since they aren't written to any logs iirc, and they should stay as developer-only logs

File: src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java [hunk 4]
```diff
@@ -61,14 +62,20 @@ public GemSmithingRecipe(ResourceLocation recipeId, Ingredient blacklist, ItemSt
...
 
+        ItemStack templateSlot = inv.getItem(0).copy();
+        ItemStack itemstack = inv.getItem(1).copy();
+        CompoundTag compoundnbt = inv.getItem(1).getTag();
+
+        MiningMaster.LOGGER.debug(templateSlot);
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java]: again, trace not debug

**Review by SwanX1 — CHANGES_REQUESTED** [review]

File: src/main/java/org/infernalstudios/miningmaster/init/MMSounds.java (hunk 2 above)
Comment (reetamb) [reply src/main/java/org/infernalstudios/miningmaster/init/MMSounds.java:32]: yes, when the Gem Forge finished the subtitle would say "eye of ender attaches"

File: src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java (hunk 3 above)
Comment (reetamb) [reply src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java]: removed the log statements since there wasn't any need to trace them either
